crates/webidl/src/first_pass.rs

```diff
@@ -40,6 +41,24 @@ pub(crate) struct InterfaceData<'src> {
     pub(crate) superclass: Option<&'src str>,
 }
 
+/// We need to collect namespace data during the first pass, to be used later.
+#[derive(Default)]
+pub(crate) struct NamespaceData<'src> {
+    /// Whether only partial namespaces were encountered
+    pub(crate) partial: bool,
+    pub(crate) operations: BTreeMap<Option<&'src str>, OperationData<'src>>,
+}
+
+impl<'src> NamespaceData<'src> {
```

Comment:
Why not just make this the implementation for `Default`?

Reply:
The reason it's the way it currently is is that I felt implementing `Default` is misleading - there isn't an 'obvious' default value for the `partial` field. I thought that having separate methods helped convey meaning.

In fact, if I had followed my own plan I would not implement Default at all, I'd have 2 methods for partial and non-partial. I'll change to this so you can see what I mean, if you don't like it I can change it to just the Default :).